syntax = "proto3";

package auth;

service AuthService {
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc ValidateToken (ValidateRequest) returns (ValidateResponse);
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string access_token = 1;
}

message ValidateRequest {
    string token = 1;
}

message ValidateResponse {
    bool valid = 1;
    string username = 2;
}